corebuild/README.md

```diff
@@ -0,0 +1,186 @@
+# ILLink.Tasks
+
+ILLink.Tasks is a package containing MSBuild tasks and targets that
+will run the linker to run during publish of a .NET Core app.
+
+ILLink.Tasks provides an MSBuild task called ILLink that makes it easy
+to run the linker from an MSBuild project file:
+
+```xml
+<ILLink AssemblyPaths="@(AssemblyFilesToLink)"
+        RootAssemblyNames="@(LinkerRootAssemblies)"
+        RootDescriptorFiles="@(LinkerRootDescriptors)"
+        OutputDirectory="output"
+        ExtraArgs="-t -c link -l none" />
+```
+
+For a description of the options that this task supports, see the
+comments in [LinkTask.cs](integration/ILLink.Tasks/LinkTask.cs).
+
+
+In addition, ILLink.Tasks contains MSBuild logic that makes the linker
+run automatically during `dotnet publish` for .NET Core apps. This
+will:
+
+- Determine the assemblies and options to pass to illink.
+- Remove unused native files from the publish output.
+- Run crossgen on the linked assemblies to improve startup performance.
+
+The full set of options is described below.
+
+## Building
+
+```
+linker> ./corebuild/dotnet.{sh/ps1} restore ./corebuild/integration/linker.sln
+linker> ./corebuild/dotnet.{sh/ps1} pack ./corebuild/integration/ILLink.Tasks/ILLink.Tasks.csproj
+```
+
+The output package will be placed in
+`corebuild/integration/bin/nupkgs`. If you are building on unix, you
+will need to adjust
+[ILLink.Tasks.nuspec](integration/ILLink.Tasks/ILLink.Tasks.nuspec). Replace
+the dll file includes with the following:
+
+`<file src="netcoreapp2.0/**/*.dll" target="tools/netcoreapp2.0" />`
+
+## Using ILLink.Tasks
+
+Add a package reference to the linker. Ensure that either the
+[dotnet-core](https://dotnet.myget.org/gallery/dotnet-core) myget feed
+or the path to the locally-built linker package path exists in the
+project's nuget.config. If using myget, you probably want to ensure
+that you're using the latest version available at
+https://dotnet.myget.org/feed/dotnet-core/package/nuget/ILLink.Tasks.
+
+After adding the package, linking will be turned on during `dotnet
+publish`. The publish output will contain the linked assemblies.
+
+## Default behavior
+
+By default, the linker will operate in a conservative mode that keeps
+all managed assemblies that aren't part of the framework (they are
+kept intact, and the linker simply copies them). It also analyzes all
+non-framework assemblies to find and keep code used by them (they are
+roots for the analysis). This means that unanalyzed reflection calls
+within the app should continue to work after linking. Reflection calls
+to code in the framework can potentially break when using the linker,
+if the target of the call is removed.
+
```

Comment:
The default action for framework assemblies is AddBypassNGen reflection won't break anything in the default mode.